src/Linter/PooledLinter.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Linter;
+
+use Amp\Parallel\Worker\DefaultPool;
+use Amp\Parallel\Worker\Environment;
+use Amp\Parallel\Worker\Task;
+
+class PooledLinter implements  LinterInterface
+{
+    private $processPool;
+
+    public function __construct()
+    {
+        $this->processPool = new DefaultPool();
```

Comment:
I would allow a `Amp\Parallel\Worker\Pool` instance to be passed via the constructor. If one isn't passed, use the one returned from `Amp\Parallel\Worker\pool()`.